src/Styleguide/Components/FollowArtistPopover/FollowArtistPopoverRow.tsx

```diff
@@ -63,7 +68,15 @@ class FollowArtistPopoverRow extends React.Component<Props, State> {
           })
         },
         updater: (store: RecordSourceSelectorProxy, data: SelectorData) => {
-          const { node } = data.followArtist.artist.related.suggested.edges[0]
+          const fetchedSuggestions =
+            data.followArtist.artist.related.suggested.edges
+          const nonSuggestedNodes = fetchedSuggestions.filter(
+            s => this.props.suggestedIds.indexOf(s.node.__id) === -1
+          )
+          const { node } = nonSuggestedNodes[0]
+          // Add new suggested artist id to suggestedIds property
+          // so that we can filter the suggested ones after next follow
+          this.props.suggestedIds.push(node.__id)
```

Comment:
Mutating props -- that's a big react nono :) Use state.

Reply:
(Also suprised this isn't throwing an error)

Reply:
It wouldn't raise an error because *in theory* it's not mutating "_the props_"

You'd need language support that says, this edits an object to make that work (swift has this)

Reply:
You could define `suggestedIds` as `ReadonlyArray<string>`

Reply:
(Ah I see, you’re probably referring to an error at runtime. Nonetheless, a compile-time error can be useful too.)

Reply:
The idea here is that I exactly want to add this new followed Id to the property that is shared with all the rows for this suggestions popup. 

I can see how that is breaking the react pattern but figured in this specific case - it's a harmless thing to do and simplifies interactions here.

So the proper way to do that would be to manage this suggestedIds state locally and dispatch an event that the parent would listen to and update it's own suggestedIds which in turn should go and update suggestedIds for all the other row children. Right? Any other options here @damassi @alloy ?

Reply:
I think you can probably use `unstated` to simplify that. Essentially React Context is what makes the most sense here, which that is a wrapper over.

A row should subscribe to that (provided by the parent), and the mutation to fetch the next suggestion can use that subscribed state to make the right query.

Reply:
I would have loved to take this case to research Relay's ability to augment the schema with local state, but as I won't be around for the next 2 weeks it might indeed be best to go with unstated.